### PR TITLE
Hotfix: Non-static method XXX::YYY should not be called statically (callable serialization issue)

### PR DESCRIPTION
**Pull request by dmitry-pro**
Hi Team! There is a weird behavior conceived to ` __METHOD__` constant when we pass it through argument in our code. I'll try to explain.
This piece of code:
https://github.com/getsentry/sentry-php/blob/master/src/Serializer/AbstractSerializer.php#L110
Guides to this method:
https://github.com/getsentry/sentry-php/blob/master/src/Serializer/AbstractSerializer.php#L213

The `callable` typehint will work incorrectly if ` __METHOD__` refers to dynamic method. It will throw an error, and then this error will spoil the original stacktrace. That's not good at all.
This snippet will show You the whole picture: http://sandbox.onlinephpfunctions.com/code/fba854e73f595f1fe78596f6bcc56371eba500b2
The error is:
```
<b>Deprecated</b>:  Non-static method AAA::get() should not be called statically in <b>[...][...]</b> on line <b>23</b><br />
```
How is it related to Sentry? If we enable Sentry for this code, and there will be an exception inside `demonstrate2`, we will have one more exception in stacktrace while serializing the stacktrace, so it will be spoiled with weird internal error. Why? Because PHP will try to execute the callable if typehint is used. No idea why.

### TL;DR: the `callable` typehint must be disabled for safety.

Thanks for attention!

### CODE REVIEW

**Comment by dmitry-pro** [comment]
> Woha, this seems really a sneaky edge case, thanks for spotting this!
> 
> Could you please add a test with this, so we're sure to not have a regression?

Ok, I'll add it shortly

**Comment by dmitry-pro** [comment]
> Woha, this seems really a sneaky edge case, thanks for spotting this!
> 
> Could you please add a test with this, so we're sure to not have a regression?

Done!

File: tests/Serializer/AbstractSerializerTest.php [hunk 1]
```diff
@@ -455,13 +456,17 @@ public function serializableCallableProvider(): array
                 'callable' => $callableWithoutNamespaces,
                 'expected' => 'Lambda void {closure} [int|null param1_70ns]',
             ],
+            [
+                'callable' => $pseudoStaticMethod,
+                'expected' => '{unserializable callable, reflection error}',
```
Comment (Jean85) [inline tests/Serializer/AbstractSerializerTest.php]: This doesn't seem an acceptable outcome...

**Comment by ste93cry** [comment]
I would say that this seems to be some strange behavior of the PHP core and that probably should be reported as a bug of the language instead of removing a typehint here. Not sure about it though

File: src/Serializer/AbstractSerializer.php [hunk 2]
```diff
@@ -210,7 +210,7 @@ protected function serializeValue($value)
      *
      * @return string
      */
-    protected function serializeCallable(callable $callable): string
+    protected function serializeCallable($callable): string
```
Comment (Jean85) [inline src/Serializer/AbstractSerializer.php]: As pointed out by @ste93cry in https://github.com/getsentry/sentry-php/pull/821#issuecomment-495261003, this is a BC, we cannot remove the type hint...

**Comment by Jean85** [comment]
Closing in favor of #821
